02-oop-inheritance/exerciseTopic2.js

```diff
@@ -0,0 +1,98 @@
+class EventEmitter {
+  constructor () {
+    this.listeners = [];
+  }
+
+  on (event, callback) {
+    if(this.listeners[event] != callback){
+      this.listeners[event] = callback
+    }
+  }
+
+  off (event) {
+    if(this.listeners[event]) this.listeners[event] = undefined;
+  }
+
+  emit (movie, event) {
+    if(this.listeners[event]) this.listeners[event](movie, event);
+  }
+}
+
+class logger {
+ constructor(){
+
+ }
+
+ log(info, functionName){
+     console.log(info.title + ' ' + functionName)
+ }
+}
+
+class classMovie extends EventEmitter{
+  constructor(title, year, duration){
+    super();
+    this.title = title;
+    this.year = year;
+    this.duration = duration;
+    this.cast = [4];
```

Comment:
You don't need to declare an array with the numbers of element it has. Here you are initializing an array and adding the number four to it.